pkg/sql/catalog/lease/lease.go

```diff
@@ -625,7 +653,9 @@ func releaseLease(ctx context.Context, lease *storedLease, m *Manager) {
 		newCtx, "sql.descriptorState: releasing descriptor lease",
 		func(ctx context.Context) {
 			m.storage.release(ctx, m.stopper, lease)
+			notify()
 		}); err != nil {
+		notify()
```

Comment:
Should `notify` actually be invoked here if the lease was not released? If so, can you document why?